AlphaWallet/Transfer/Controllers/TransactionConfigurator.swift

```diff
@@ -155,11 +155,11 @@ class TransactionConfigurator {
 
 // swiftlint:disable function_body_length
     func load(completion: @escaping (ResultResult<Void, AnyError>.t) -> Void) {
-        if gasLimitNotSet {
-            estimateGasLimit()
-        }
         switch transaction.transferType {
         case .dapp:
+            if gasLimitNotSet {
+                estimateGasLimit()
+            }
```

Comment:
Should it still do an early return like before #1926 is applied or is removing the early return intentional?

```
if gasLimitNotSet {
    estimateGasLimit()
} else {
    completion(.success(()))
}
```

Reply:
Nope. In the case of dapp, it usually has the gas limit but it never sets the gas price properly (defaults to 9 gwei). The other cases can get the gas price properly.